### PR TITLE
fix(#69): add new class annotations to deprecated classes

### LINKED ISSUE #69: Decorate deprecations with suggested replacements
Please decorate the deprecations with suggested replacements.
https://github.com/open-telemetry/semantic-conventions-java/pull/62

### PR DESCRIPTION
**Pull request by cboudereau**
fixes #69

### CODE REVIEW

**Comment by cboudereau** [comment]
Hi, 

I think the PR is done.

I used @see instead of @link since the moved classes are in another jar which is not referenced.

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java [hunk 1]
```diff
@@ -25,7 +25,12 @@ public final class ResourceAttributes {
   /** The URL of the OpenTelemetry schema for these keys and values. */
   @Deprecated public static final String SCHEMA_URL = "https://opentelemetry.io/schemas/1.23.1";
 
-  /** The cloud account ID the resource is assigned to. */
+  /**
+   * The cloud account ID the resource is assigned to.
+   *
+   * @deprecated deprecated in favor of stable {@see
+   *     io.opentelemetry.semconv.incubating.CloudIncubatingAttributes#CLOUD_ACCOUNT_ID} attribute.
```
Comment (trask) [inline semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java:32]: (the incubating attributes aren't stable)
```suggestion
   * @deprecated deprecated in favor of {@see
   *     io.opentelemetry.semconv.incubating.CloudIncubatingAttributes#CLOUD_ACCOUNT_ID} attribute.
```

**Comment by cboudereau** [comment]
@trask , it should be ok but the markdown link check failed because all "maven-badges.herokuapp.com" badges are down.

This is a problem related to maven-badges.herokuapp.com itself though.

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java [hunk 2]
```diff
@@ -1013,242 +1514,649 @@ public final class ResourceAttributes {
...
 
   // Enum definitions
   @Deprecated
   public static final class CloudPlatformValues {
-    /** Alibaba Cloud Elastic Compute Service. */
+    /**
+     * Alibaba Cloud Elastic Compute Service.
+     *
+     * @deprecated deprecated in favor of {@see
+     *     io.opentelemetry.semconv.incubating.AlibabaIncubatingAttributes.CloudPlatformValues#ALIBABA_CLOUD_ECS}
```
Comment (trask) [inline semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java:1542]: sorry, these inner class names have just changed in #83

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java [hunk 3]
```diff
@@ -971,40 +1432,80 @@ public final class ResourceAttributes {
...
+   */
   @Deprecated
   public static final AttributeKey<String> OTEL_SCOPE_VERSION = stringKey("otel.scope.version");
 
   /**
    * Deprecated, use the {@code otel.scope.name} attribute.
    *
    * @deprecated Deprecated, use the `otel.scope.name` attribute.
+   * @deprecated deprecated in favor of {@see
+   *     io.opentelemetry.semconv.incubating.OtelIncubatingAttributes#OTEL_LIBRARY_NAME} attribute.
```
Comment (trask) [inline semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java:1508]: duplicate `@deprecated` here, maybe combine them somehow?

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java [hunk 4]
```diff
@@ -1013,242 +1514,649 @@ public final class ResourceAttributes {
    * Deprecated, use the {@code otel.scope.version} attribute.
    *
    * @deprecated Deprecated, use the `otel.scope.version` attribute.
+   * @deprecated deprecated in favor of {@see
+   *     io.opentelemetry.semconv.incubating.OtelIncubatingAttributes#OTEL_LIBRARY_VERSION}
+   *     attribute.
```
Comment (trask) [inline semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java:1519]: and here

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java (hunk 2 above)
Comment (cboudereau) [reply semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java]: Done. I renamed all occurences to the CloudIncubatingAttributes

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java (hunk 4 above)
Comment (cboudereau) [reply semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java]: done

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java (hunk 3 above)
Comment (cboudereau) [reply semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java]: done, I also checked for all duplicates in a commit

**Comment by cboudereau** [comment]
Hi @trask, I updated the PR to merge the recent changes from #83. It should be ok 👍

File: semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java [hunk 5]
```diff
@@ -971,284 +1432,733 @@ public final class ResourceAttributes {
...
+     *     attribute.
+     */
     @Deprecated public static final String EC2 = "ec2";
 
-    /** fargate. */
+    /**
+     * fargate.
+     *
+     * @deprecated deprecated in favor of {@see
+     *     io.opentelemetry.semconv.incubating.AwsIncubatingAttributes.AwsEcsLaunchtypeValues#FARGATE}
```
Comment (trask) [inline semconv/src/main/java/io/opentelemetry/semconv/ResourceAttributes.java:1865]:
```suggestion
     *     io.opentelemetry.semconv.incubating.AwsIncubatingAttributes.AwsEcsLaunchtypeIncubatingValues#FARGATE}
```

**Review by trask — APPROVED** [review]
thanks!

**Comment by cboudereau** [comment]
> thanks!

I am not authorized to merge though.

**Comment by trask** [comment]
it will need to be merged by one of the @open-telemetry/java-maintainers

**Pull request merged**